src/stream/src/executor/now.rs

```diff
@@ -79,7 +79,6 @@ impl<S: StateStore> NowExecutor<S> {
             if !initialized {
                 // Handle the first barrier.
                 state_table.init_epoch(barrier.epoch);
-
                 let state_row = {
                     let data_iter = state_table.iter_row(Default::default()).await?;
```

Comment:
The semantics of `iter_row` looks unclear to me. It seems that it **should** iterate over all rows in the table, in spite of the properties of the table. However, **in fact**, it only works for tables without dist keys, on which when we iterate the `pk_prefix` can, actually should, be empty row.

What makes the method more unreasonable is that, many places where we call `iter_row_with_pk_range` (after this PR, `vnode_iter_row`), we meant to iterate over the whole table or a range of it across all its vnodes. That means, the caller will iterate over all related vnodes, and call `iter_row_with_pk_range` with each of these vnodes with a same range.

So, either one of the following will make more sense to me:

- Simply don't provide `iter_row` method;
- Iterate over vnodes inside `iter_row` so that it works for tables with dist keys, and, provide another version of `iter_row_with_pk_range` that also iterate over vnode internally, without need to pass in `vnode` argument. The existing `iter_row_with_pk_range` may still be needed by some executors that need to carefully handle vnode concept, but that's not a big problem.